shared/src/main/scala/coop/rchain/pcasper/finalization/package.scala

```diff
@@ -0,0 +1,39 @@
+package coop.rchain.pcasper
+
+import cats.effect.Sync
+import cats.syntax.all._
+
+package object finalization {
+
+  // Fringe going all across boded senders, or level of a partition
+  type Fringe[M, S] = Map[S, M]
+
+  def isSupermajority[S](senders: Iterable[S], bondsMap: Map[S, Long]): Boolean =
+    senders.nonEmpty && {
+      val totalStake   = bondsMap.keysIterator.map(bondsMap).sum
+      val sendersStake = senders.toIterator.map(bondsMap.getOrElse(_, 0L)).sum
+      sendersStake * 3 >= totalStake * 2
+    }
+
+  def totalStake[S](bonds: Map[S, Long]): Long = bonds.valuesIterator.sum
+
+  def highestMessages[M, S](views: Iterable[Map[S, M]])(seqNum: M => Long): Map[S, M] =
```

Comment:
The type `Map[S, M]` is used often in the code, but it's not clear to understand what it represents. Consider introducing type alias/aliases for the type.